cluster/cluster_interface.ml

```diff
@@ -14,12 +14,26 @@ type debug_info = string [@@deriving rpcty]
 (** Name of the cluster *)
 type cluster_name = string [@@deriving rpcty]
 
-(** An IPv4 address (a.b.c.d) *)
-type address = IPv4 of string [@@deriving rpcty]
+type ip_with_hostname = {ip: string; hostname: string} [@@deriving rpcty]
```

Comment:
Can we use a [`Ipaddr.t`](https://github.com/mirage/ocaml-ipaddr) here instead of a string?